client/components/training-session.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import moment from 'moment';
+
+export default class TrainingSession extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      trainingSession: [],
+      startDate: this.props.date
+    };
+  }
+
+  async componentDidMount() {
+    const date = moment(this.state.startDate).format();
+    const sessionDate = date.slice(0, 10);
+    const response = await fetch(`/api/training/${sessionDate}`);
+    const trainingSession = await response.json();
+    this.setState({ trainingSession });
+  }
+
+  async componentDidUpdate(prevProps) {
+    if (this.props.date !== prevProps.date) {
+      await this.setState({
+        startDate: this.props.date
+      });
+      const date = moment(this.state.startDate).format();
+      const sessionDate = date.slice(0, 10);
+      fetch(`/api/training/${sessionDate}`)
+        .then(res => res.json())
+        .then(trainingSession => {
+          this.setState({ trainingSession });
+        });
+    }
+  }
+
+  render() {
+    const session = this.state.trainingSession;
+    if (!session) {
+      return null;
+    }
```

Comment:
This guard shouldn't be necessary if your initial `trainingSession` is an array. It's safe to call `.map()` on an empty array, it just gives you another empty array.